### PR TITLE
feature/#531 expose get_cycle_scenarios API

### PR DESCRIPTION
**Pull request by toan-quach**
#531

### CODE REVIEW

File: src/taipy/core/taipy.py [hunk 1]
```diff
@@ -613,3 +613,14 @@ def update_parent_dict(parents_set, parent_dict, key):
             get_parents(parent, parent_dict)
 
     return parent_dict
+
+
+def get_cycle_scenarios() -> Dict[Optional[Cycle], List[Scenario]]:
+
```
Comment (trgiangdo) [inline src/taipy/core/taipy.py:629]: A docstring here would be helpful.

File: src/taipy/core/taipy.py [hunk 2]
```diff
@@ -613,3 +613,14 @@ def update_parent_dict(parents_set, parent_dict, key):
...
+    cycles_scenarios: Dict[Optional[Cycle], List[Scenario]] = {}
+    for scenario in get_scenarios():
```
Comment (trgiangdo) [inline src/taipy/core/taipy.py:631]: Question: Is the list returned by `get_scenarios()` sorted?

If not, should it sorted by creation date or something?
If yes, then great 😄

File: tests/test_taipy.py [hunk 3]
```diff
@@ -497,3 +497,38 @@ def assert_result_parents_and_expected_parents(parents, expected_parents):
...
+            scenario_4.cycle: [scenario_4.id],
+            None: [scenario_5_1.id, scenario_5_2.id, scenario_5_3.id],
+        }
+
+        cycles_scenarios = tp.get_cycle_scenarios()
+
+        assert expected_cycles_scenarios.keys() == cycles_scenarios.keys()
+        for cycle, scenarios in cycles_scenarios.items():
+            expected_scenarios = expected_cycles_scenarios[cycle]
+            assert sorted([scenario.id for scenario in scenarios]) == sorted(expected_scenarios)
```
Comment (trgiangdo) [inline tests/test_taipy.py:540]: Is `sort()` necessary here?

**Review by trgiangdo — COMMENTED** [review]
I think we should at least add a docstring. Otherwise, LGTM!

I also have some minor questions.

File: src/taipy/core/taipy.py (hunk 2 above)
Comment (toan-quach) [reply src/taipy/core/taipy.py:631]: It iterates through the .data scenario folder for filesystem so there's probably some form of sort but I'm not sure :/

File: tests/test_taipy.py (hunk 3 above)
Comment (toan-quach) [reply tests/test_taipy.py:540]: yep, since I don't know the order of the list that will come out, a sort is needed :(

File: tests/test_taipy.py [hunk 4]
```diff
@@ -497,3 +497,38 @@ def assert_result_parents_and_expected_parents(parents, expected_parents):
...
+            [],
+            Frequency.DAILY,
+        )
+        scenario_cfg_2 = Config.configure_scenario("s2", [], Frequency.WEEKLY)
+        scenario_cfg_3 = Config.configure_scenario("s3", [], Frequency.MONTHLY)
+        scenario_cfg_4 = Config.configure_scenario("s4", [], Frequency.YEARLY)
+        scenario_cfg_5 = Config.configure_scenario("s5", [], None)
+
+        scenario_1_1 = tp.create_scenario(scenario_cfg_1, datetime.datetime.now())
+        scenario_1_2 = tp.create_scenario(scenario_cfg_1, datetime.datetime.now())
```
Comment (jrobinAV) [inline tests/test_taipy.py:513]: Can you also test with the following?
```suggestion
        scenario_1_2 = tp.create_scenario(scenario_cfg_1, datetime.datetime.now())
        scenario_1_3 = tp.create_scenario(scenario_cfg_1, datetime.datetime.now() + datetime.timedelta(days=1))
        scenario_1_4 = tp.create_scenario(scenario_cfg_1, datetime.datetime.now()  + datetime.timedelta(days=8))
        scenario_1_5 = tp.create_scenario(scenario_cfg_1, datetime.datetime.now() + datetime.timedelta(days=25))
```

File: src/taipy/core/taipy.py (hunk 2 above)
Comment (jrobinAV) [reply src/taipy/core/taipy.py:631]: Thats a good question. We explicitly stated with Fabien that the first version does not require any sort.

**Review by jrobinAV — DISMISSED** [review]

File: src/taipy/core/taipy.py [hunk 5]
```diff
@@ -613,3 +613,19 @@ def update_parent_dict(parents_set, parent_dict, key):
             get_parents(parent, parent_dict)
 
     return parent_dict
+
+
+def get_cycles_scenarios() -> Dict[Optional[Cycle], List[Scenario]]:
+    """Get the cycles and its scenarios.
```
Comment (jrobinAV) [inline src/taipy/core/taipy.py]:
```suggestion
    """Get the scenarios grouped by cycles.
```

File: src/taipy/core/taipy.py [hunk 6]
```diff
@@ -613,3 +613,19 @@ def update_parent_dict(parents_set, parent_dict, key):
...
+
+    Returns:
+        The dictionary of all cycles and its corresponding scenarios.
```
Comment (jrobinAV) [inline src/taipy/core/taipy.py]:
```suggestion
        The dictionary of all cycles and their corresponding scenarios.
```

**Review by jrobinAV — DISMISSED** [review]

**Review by trgiangdo — APPROVED** [review]

**Pull request merged**